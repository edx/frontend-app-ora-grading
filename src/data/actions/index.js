import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';
import submissions from './submissions';

export default StrictDict({
  app,
  grading,
  submissions,
});
