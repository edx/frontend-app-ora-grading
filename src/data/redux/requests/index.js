export { actions, reducer } from './reducer';
