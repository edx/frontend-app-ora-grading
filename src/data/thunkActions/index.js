import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';

export default StrictDict({
  app,
  grading,
});
