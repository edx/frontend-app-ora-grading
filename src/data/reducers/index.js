import { combineReducers } from 'redux';

import app from './app';
import submissions from './submissions';

/* istanbul ignore next */
const rootReducer = combineReducers({
  app,
  submissions,
});

export default rootReducer;
