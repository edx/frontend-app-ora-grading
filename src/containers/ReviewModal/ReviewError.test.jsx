import React from 'react';
import { shallow } from 'enzyme';

import { thunkActions } from 'data/redux';

import {
  ReviewError,
  mapDispatchToProps,
} from './ReviewError';

let el;
jest.useFakeTimers('modern');

describe('ReviewError component', () => {
  const props = {};
  describe('component', () => {
    beforeEach(() => {
      props.reload = jest.fn();
    });
    describe('render tests', () => {
      beforeEach(() => {
        el = shallow(<ReviewError {...props} />);
      });
      test('snapshot', () => {
        expect(el).toMatchSnapshot();
      });
    });
  });
  describe('mapDispatchToProps', () => {
    it('loads reload from thunkActions.grading.reloadSubmission', () => {
      expect(mapDispatchToProps.reload).toEqual(thunkActions.grading.loadSubmission);
    });
  });
});
