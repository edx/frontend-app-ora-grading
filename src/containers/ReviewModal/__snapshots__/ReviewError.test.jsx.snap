// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewError component component render tests snapshot 1`] = `
<Alert
  actions={
    Array [
      <Button
        onClick={[MockFunction]}
      >
        <FormattedMessage
          defaultMessage="Reload submission"
          description="Reload button text in case of network failure"
          id="ora-grading.ReviewModal.reloadSubmission"
        />
      </Button>,
    ]
  }
  variant="danger"
>
  <Alert.Heading>
    <FormattedMessage
      defaultMessage="Error loading submissions"
      description="Submission response load failure alert header"
      id="ora-grading.ReviewModal.loadErrorHeading"
    />
  </Alert.Heading>
  <p>
    <FormattedMessage
      defaultMessage="An error occurred while loading this submission.  Try reloading this submission."
      description="Submission response load failure alert message"
      id="ora-grading.ReviewModal.loadErrorMessage1"
    />
  </p>
</Alert>
`;
